go/ct/gen/transaction_context_test.go

```diff
@@ -33,3 +93,186 @@ func TestTransactionContext_Generate(t *testing.T) {
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+			},
+			shouldBePresent: []Variable{"v1"},
+		},
+		"present-and-assigned": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+				(assignment)[Variable("v1")] = common.NewU256(5)
+			},
+			shouldBePresent: []Variable{"v1"},
+		},
+		"absent": {
+			setup: func(txCtxGen *TransactionContextGenerator, _ Assignment) {
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v1"))
+			},
+			shouldBeAbsent: []Variable{"v1"},
+		},
+		"absent-and-assigned": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v1"))
+				(assignment)[Variable("v1")] = common.NewU256(5)
+			},
+			shouldBeAbsent: []Variable{"v1"},
+		},
+		"absent-and-present": {
+			setup: func(txCtxGen *TransactionContextGenerator, _ Assignment) {
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v2"))
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+			},
+			shouldBePresent: []Variable{"v1"},
+			shouldBeAbsent:  []Variable{"v2"},
+		},
+		"present-and-absent": {
+			setup: func(txCtxGen *TransactionContextGenerator, _ Assignment) {
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v2"))
+			},
+			shouldBePresent: []Variable{"v1"},
+			shouldBeAbsent:  []Variable{"v2"},
+		},
+		"present-assigned-and-absent": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+				(assignment)[Variable("v1")] = common.NewU256(5)
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v2"))
+			},
+			shouldBePresent: []Variable{"v1"},
+			shouldBeAbsent:  []Variable{"v2"},
+		},
+		"absent-assigned-and-present": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v2"))
+				txCtxGen.IsPresentBlobHashIndex(Variable("v1"))
+				(assignment)[Variable("v2")] = common.NewU256(5)
+			},
+			shouldBePresent: []Variable{"v1"},
+			shouldBeAbsent:  []Variable{"v2"},
+		},
+		"assigned-too-big-absent": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				(assignment)[Variable("v1")] = common.NewU256(1, 1)
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v1"))
+			},
+			shouldBeAbsent: []Variable{"v1"},
+		},
+		"zero-assigned-absent": {
+			setup: func(txCtxGen *TransactionContextGenerator, assignment Assignment) {
+				(assignment)[Variable("v1")] = common.NewU256(0)
+				txCtxGen.IsAbsentBlobHashIndex(Variable("v1"))
+			},
+			shouldBeAbsent: []Variable{"v1"},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			rnd := rand.New(0)
+			txCtxGen := NewTransactionContextGenerator()
+			assignment := Assignment{}
+			test.setup(txCtxGen, assignment)
+			assignmentBackup := assignment
```

Comment:
The assignment is a `map`, and thus a reference type. The backup you create here is just an alias of the `assignment`, and all changes to the `assignment` are also affecting the backup.

To make a real backup, use `maps.Clone()`.